cardano-cli/src/Cardano/CLI/Shelley/Key.hs

```diff
@@ -231,6 +234,33 @@ deserialiseInputAnyOf bech32Types textEnvTypes inputBs =
         -- The input was valid Bech32, but some other error occurred.
         Left err -> DeserialiseInputError $ InputBech32DecodeError err
 
+------------------------------------------------------------------------------
+-- Formatted/encoded output serialisation
+------------------------------------------------------------------------------
+
+-- | Where to write some output.
+data OutputDirection
+  = OutputDirectionStdout
+  -- ^ Write output to @stdout@.
+  | OutputDirectionFile !FilePath
+  -- ^ Write output to a provided file.
+  deriving (Eq, Show)
+
+serialiseInputToBech32AndWrite
+  :: SerialiseAsBech32 a
+  => OutputDirection
```

Comment:
I think we should stick with the `Maybe OutputFile` pattern we've been using. This means we don't need `OutputDirection` and therefore we have one less type to think about.

Reply:
I'm not sure `Maybe OutputFile` properly describes the intention here. 

e.g. `serialiseInputToBech32AndWrite Nothing someInput` doesn't really describe that we're intending to write to `stdout`.

Reply:
> I can continue using `Maybe OutputFile` in the CLI parsers, if you like, and just convert it to `OutputDirection` when calling `serialiseInputToBech32AndWrite`.

I've gone ahead and done this. Let me know your thoughts.